File: GoTest/tmp.go
/*
 * @Author: NEFU AB-IN
 * @Date: 2023-08-06 22:57:54
 * @FilePath: \GoTest\tmp.go
 * @LastEditTime: 2023-08-06 23:05:28
 */

package main

import (
	"bufio"
	"fmt"
	"os"
)

var (
	in  = bufio.NewReader(os.Stdin)
	out = bufio.NewWriter(os.Stdout)
)

func Read[T any]() T {
	var i T
	fmt.Fscan(in, &i)
	return i
}

const N int = 1e5 + 10

func solve() {

}

func main() {
	defer out.Flush()
	T := Read[int]()
	for T > 0 {
		solve()
		T--
	}
}

File: ACM/Acwing/TMP/TMP.py
'''
Author: NEFU AB-IN
Date: 2023-09-09 00:26:07
FilePath: \Acwing\TMP\TMP.py
LastEditTime: 2023-09-09 19:45:06
'''
import re
while True:
    try:
        s1 = input().lower()
        s2 = input().lower()
        s1 = s1.replace('.', '\.').replace('?', '[0-9a-z]').replace('*','#')
        s1 = re.sub('#+','[0-9a-z]*',s1)
        if bool(re.fullmatch(s1, s2)):
            print('true')
        else:
            print('false')
    except:
        break
